proxy/src/main/java/net/md_5/bungee/util/BungeeTranslationUtil.java

```diff
@@ -0,0 +1,146 @@
+package net.md_5.bungee.util;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.md_5.bungee.api.chat.BaseComponent;
+import net.md_5.bungee.api.chat.ComponentBuilder;
+import net.md_5.bungee.api.chat.TextComponent;
+import net.md_5.bungee.api.chat.TranslatableComponent;
+import org.jetbrains.annotations.NotNull;
+
+public class BungeeTranslationUtil
+{
+    // support only simple translations, no escaping and ignore specialized formatting
+    private static final Pattern TRANSLATE_FORMAT = Pattern.compile( "\\{(\\d*)(?:,[^}]+)?}" );
+
+    public static @NotNull BaseComponent getTranslationComponent0(String trans, Object... args)
+    {
+
+        Matcher matcher = TRANSLATE_FORMAT.matcher( trans );
+        int position = 0;
+        int i = 0;
+        BaseComponent lastNonArg = null;
+        ComponentBuilder builder = new ComponentBuilder();
+        while ( matcher.find() )
+        {
+            // Add the static part of the string
+            int pos = matcher.start();
+            if ( pos != position )
+            {
+                builder.appendLegacy( trans.substring( position, pos ) );
+                BaseComponent curr = builder.getCurrentComponent();
+                if ( lastNonArg != null )
+                {
+                    // undo eventual formatting of previous argument
+                    boolean colorChange = curr.isReset();
+                    if ( !Objects.equals( lastNonArg.getColor(), curr.getColor() ) )
+                    {
+                        colorChange = true;
+                        builder.color( lastNonArg.getColor() );
+                    }
+                    if ( colorChange && lastNonArg.isBold() || lastNonArg.isBold() != curr.isBold() )
```

Comment:
Could setStyle simplify all these ifs?

Reply:
I'll need to look in detail how the componentbuilder is doing stuff withformat retention and nesting components esp. in conjunction with appendLegacy. The way I wrote this, it adds fewest formatting changes due to the style getters involve questioning the parent. An eventual null in a component's style might mean "white", while when appended/nested, the ComponentBuilder will likely make it mean "keep previous color",
I could add it with no format retention and then use setStyle if you don't like this, but I would prefer to extract this maybe into a separate function.

Maybe its better to use no format retention + setStyle, will check out eventual behaviour differences.